source/System.cpp

```diff
@@ -915,7 +915,7 @@ void System::LoadObject(const DataNode &node, Set<Planet> &planets, int parent)
 }
 
 
-
+// This looks relevant to setting if an object is a star, but it has no explanation.
```

Comment:
As above.
What do you want to know about this method?

Reply:
Already removed. That being said, I was wondering what it does, since, well, there's no comments as to what it does.

Reply:
It parses a DataNode for properties of a StellarObject and fills in the StellarObject it is passed with them.

Reply:
... I only vaguelly understand that, but thank you anyway.

Reply:
`DataNode` is the structure the game converts the data in the data files to.
Here's the comment on it:
```
// A DataNode is a single line of a DataFile. It consists of one or more tokens,
// which can be interpreted either as strings or as floating point values, and
// it may also have "children," which may each in turn have their own children.
// The tokens of a node are separated by white space, with quotation marks being
// used to group multiple words into a single token. If the token text contains
// quotation marks, it should be enclosed in backticks instead.
```

In this case, `parse` refers to the process of looking at each of the children of the DataNode (named `node`) which should be:
`object "optional planet name"`
To find relevant data.
For example, if the first token of one of the children is `sprite`, and that node has additional tokens, it will pass that node to a method for loading a sprite. (I believe this method is in the Body class.)